### PR TITLE
plugins.auftanken: new plugin

### LINKED ISSUE #3602: https://www.auftanken.tv/
## Plugin Request

- [x] This is a plugin request and I have read the contribution guidelines and plugin request requirements.
- [ ] I am using the latest development version from the master branch.

### Example stream URLs

1. https://tvhosa-pez0302.sctv.ch/http_adaptive_streaming/GEMBFJAABFBBBEAG.m3u8
2. ...
3. ...

### Additional comments, etc.
site uses tokens, so example url will prolly change very soon.
swiss channel, works with non .ch ip too

[Love Streamlink? Please consider supporting our collective. Thanks!](https://opencollective.com/streamlink/donate)

### PR DESCRIPTION
**Pull request by Billy2011**
closes #3602

### CODE REVIEW

File: src/streamlink/plugins/auftanken.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import re
+
+from streamlink.plugin import Plugin
+from streamlink.stream import HLSStream
+
+log = logging.getLogger(__name__)
+
+
+class AuftankenTV(Plugin):
+    _url_re = re.compile(r"https?://(?:www\.)?auftanken.tv/livestream/?")
```
Comment (bastimeyer) [inline src/streamlink/plugins/auftanken.py:11]:
```suggestion
    _url_re = re.compile(r"https?://(?:www\.)?auftanken\.tv/livestream/?")
```

**Review by bastimeyer — COMMENTED** [review]
The plugin itself looks fine, and it's rather simple, but, and this has just been discussed earlier today / yesterday in another thread, this site is the pure definition of niche. Just take a look at their target audience, the program being broadcasted and their social media presence, eg. on Twitter with a whopping follower number of one. Which value does a plugin for this site bring to Streamlink? The plugin requests also doesn't give any additional information about it.

**Comment by Billy2011** [comment]
> Which value does a plugin for this site bring to Streamlink?

I don’t think you can answer that in a general way. This channel offers a varied program for anyone who is not only interested in gaming, news, etc. Should streamlink only be there for the broad masses?

**Comment by beardypig** [comment]
@Billy2011 well said. But each new plugin slows down the loading process (which really only affects slower machines) and adds maintenance burden. 

<offtopic>
Loading times could be improved by changing a few things though. All of the plugins compile the regexs they might only use once, and if the plugin is used - but they are still compiled on import. So of the URL regexs could be replaced by simple substring matches, etc. We could introduce a cache of previously used URLs and which plugins they matched. etc. etc.
</offtopic>

**Comment by Billy2011** [comment]
> But each new plugin slows down the loading process (which really only affects slower machines) and adds maintenance burden.

Then you should limit the number of plugins, because this is the only way to avoid subjective reviews. For me, social "media presence" is also not an objective criterion, unless only the supposed mass taste counts, which is propagated somewhere.

**Comment by bastimeyer** [comment]
My point was that this kind of niche-content that's being shown there in addition to the state of their social media presence is simply an indicator for their viewer numbers, because the site itself doesn't show it. I don't care about the actual content.

Adding a plugin to Streamlink which will probably only be used by at most one person from time to time does not bring any value to the overall project, because as @beardypig already said, it would unnecessarily slow down the entire application due to its current plugin resolving logic. This might not be much on machines that are not slow, but it also adds maintenance burden for something that is probably not being used anyway.

And on top of that, the future of smaller channels / sites like that is always unknown and if they go offline / bankrupt, then we have to remove the plugin again. This has already been the case dozens of times for similar sites and the offline-report came years later after they went down, because nobody noticed it, which is a good indicator of how often niche-plugins are used in this project.

As @gravyboat said in https://github.com/streamlink/streamlink/issues/3334#issuecomment-792076010, we need to find a solution for this, so that niche-plugins can exist but don't disrupt the overall project.

**Pull request closed without merging**